frame/staking/src/lib.rs

```diff
@@ -1164,23 +1017,73 @@ decl_storage! {
+			type PalletPrefix: Get<&'static str>;
+		}
+
+		// TODO: complete this
+		macro_rules! generate_storage_types {
+			($name:ident<T: $trait:tt> => Map<$key:ty, $value:ty>) => {
+				unreachable!()
+			};
+			($name:ident<T: $trait:tt> => Value<$value:ty>) => {
+				paste::paste! {
+					struct [<$name Instance>]<T>($crate::sp_std::marker::PhantomData<T>);
+					impl<T: $trait> $crate::frame_support::traits::StorageInstance for [<$name Instance>]<T> {
+						fn pallet_prefix() -> &'static str {
+							T::PalletPrefix::get()
+						}
+						const STORAGE_PREFIX: &'static str = stringify!($name);
+					}
+					type $name<T> = StorageValue<[<$name Instance>]<T>, $value, ValueQuery>;
+				}
 			}
 		}
-	}
-	pub fn migrate_to_blockable<T: Config>() -> frame_support::weights::Weight {
-		Validators::<T>::translate::<OldValidatorPrefs, _>(|_, p| Some(p.upgraded()));
-		ErasValidatorPrefs::<T>::translate::<OldValidatorPrefs, _>(|_, _, p| Some(p.upgraded()));
-		T::BlockWeights::get().max_block
+
+		// NOTE: value type ain't matter, we just set it to () here.
+		generate_storage_types!(SnapshotValidators<T: V6Config> => Value<()>);
+		generate_storage_types!(SnapshotNominators<T: V6Config> => Value<()>);
+		generate_storage_types!(QueuedElected<T: V6Config> => Value<()>);
+		generate_storage_types!(QueuedScore<T: V6Config> => Value<()>);
+		generate_storage_types!(EraElectionStatus<T: V6Config> => Value<()>);
+		generate_storage_types!(IsCurrentSessionFinal<T: V6Config> => Value<()>);
+
+		/// check to execute prior to migration.
+		pub fn pre_migration<T: V6Config>() -> Result<(), &'static str> {
+			ensure!(SnapshotValidators::<T>::exists(), "SnapshotValidators storage item not found!");
+			ensure!(SnapshotNominators::<T>::exists(), "SnapshotNominators storage item not found!");
+			ensure!(QueuedElected::<T>::exists(), "QueuedElected storage item not found!");
+			ensure!(QueuedScore::<T>::exists(), "QueuedScore storage item not found!");
+			ensure!(EraElectionStatus::<T>::exists(), "EraElectionStatus storage item not found!");
+			ensure!(IsCurrentSessionFinal::<T>::exists(), "IsCurrentSessionFinal storage item not found!");
+			Ok(())
+		}
+
+		/// Migrate storage to v6.
+		pub fn migrate<T: V6Config>() -> Weight {
+			if StorageVersion::get() == Releases::V5_0_0 {
+				log!(info, "Migrating staking to Releases::V6_0_0");
+
+				SnapshotValidators::<T>::kill();
+				SnapshotNominators::<T>::kill();
+				QueuedElected::<T>::kill();
+				QueuedScore::<T>::kill();
+				EraElectionStatus::<T>::kill();
+				IsCurrentSessionFinal::<T>::kill();
+
+				StorageVersion::put(Releases::V6_0_0);
+			}
+			T::DbWeight::get().writes(6)
+		}
+		// TODO: TODO: companion to call this, also, this needs some consideration: When will this
+		// runtime land? if it happens very close to an epoch change?
+		// TODO: notify jacogr that this limitation no longer exists.
+		// TODO: tx-version shall be bump.
```

Comment:
Which of these TODOs remain relevant?

Reply:
All removed and appropriate issues created.